Add light mode support

martin-e91 opened the pull request:
# Description

Added the support to the Light Mode

Fixes #75

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [X] Manually tested

**Configuration**:
OS and Version: macOS 11.2.3

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

# Screenshots
<img src="https://user-images.githubusercontent.com/24659391/115833138-fcf23e00-a413-11eb-87b1-47d3ad3d78d3.png" height=500>
<img src="https://user-images.githubusercontent.com/24659391/115833150-ff549800-a413-11eb-80a4-2898e6b911a0.png" height=500>
<img src="https://user-images.githubusercontent.com/24659391/115833157-ffed2e80-a413-11eb-9e55-7dadac462dca.png" height=500>
<img src="https://user-images.githubusercontent.com/24659391/115833159-011e5b80-a414-11eb-8de5-8ffe397271e0.png" height=500>

FabrizioBrancati commented on Sources/App/Views/Sidebar/SidebarItem.swift:
```diff
@@ -37,7 +47,7 @@ struct SidebarItem: View {
           Text(section.title)
```
It's better to set the `foregroundColor` to all the `Text`s, `TextField`s, etc too 😄

TheInkedEngineer commented:
@martin-e91 can you please add screenshots?

martin-e91 commented:
PR updated 👍🏾 
@FabrizioBrancati @TheInkedEngineer

FabrizioBrancati commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -25,7 +25,6 @@ If applicable, add screenshots to help explain your problem.
 
 **Configuration**
 OS and Version: [e.g. macOS 11.2]
-App Version: [e.g. 0.1.0]
```
Why do you want to remove this? 😅

martin-e91 replied:
Fixed

FabrizioBrancati commented on .github/pull_request_template.md:
```diff
@@ -14,14 +14,13 @@ Please delete options that are not relevant.
 # How Has This Been Tested?
 
-Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
 
 - [ ] Test A
 - [ ] Test B
 
 **Configuration**:
 OS and Version: [e.g. macOS 11.2]
-App Version: [e.g. 0.1.0]
```
There 2 too

martin-e91 replied:
Fixed

FabrizioBrancati commented on Sources/App/Views/Common/Buttons/AccentButtonStyle.swift:
```diff
@@ -2,23 +2,22 @@
+  /// - Parameter colorScheme: The color scheme of the app.
   /// - Parameter isEnabled: Whether or not the button is enabled.
-  init(isEnabled: Bool = true) {
+  init(colorScheme: ColorScheme, isEnabled: Bool = true) {
+    self.colorScheme = colorScheme
```
Is this really necessary?
Can you access to `@Environment(\.colorScheme)` from here?

martin-e91 replied:
I can't because it wouldn't be injected in `AccentButtonStyle` as it is not a `View`

FabrizioBrancati commented on Sources/App/Views/Common/EmptyState.swift:
```diff
@@ -26,8 +26,10 @@ struct EmptyState: View {
         .frame(maxWidth: 45, maxHeight: 45)
         .padding()
         .font(.body)
+        .foregroundColor(.ristretto)
       Text(text)
         .font(.body)
+        .foregroundColor(.ristretto)
```
Why `ristretto`?

martin-e91 replied:
It's the color used on the Figma

FabrizioBrancati commented on Sources/App/Views/Sections/Editor/Detail/EditorDetail.swift:
```diff
@@ -9,6 +9,9 @@ struct EditorDetail: View {
   // MARK: - Stored Properties
 
+  /// The current color scheme of the app.
+  @Environment(\.colorScheme) var colorScheme: ColorScheme
```
Check if this is really needed

martin-e91 replied:
Unfortunately yes. I was looking into a way for centralizing it. 
I tried to move this into an `extension` but it didn't work as that way because `colorScheme` is not injected in the `View`.

FabrizioBrancati commented on Sources/App/Views/Sections/Server/Detail/Info/ServerDetailInfo.swift:
```diff
@@ -11,21 +11,33 @@ struct ServerDetailInfo: View {
   // MARK: - Stored Properties
 
+  /// The current color scheme of the app.
+  @Environment(\.colorScheme) var colorScheme: ColorScheme
```
Here too

martin-e91 replied:
Same as above

FabrizioBrancati commented on Sources/App/Views/Sidebar/SidebarItem.swift:
```diff
@@ -9,6 +9,9 @@ struct SidebarItem: View {
   // MARK: - Stored Properties
 
+  /// The color scheme currently selected by the user.
+  @Environment(\.colorScheme) var colorScheme: ColorScheme
```
Why do you need it?

martin-e91 replied:
Same as above

martin-e91 commented:
Updated PR: 
- Removed `colorScheme` where used as `Environment` property in order to reduce verbosity.
- Added `milk` `Color` which is universal for both light and dark mode.

davideruisi commented on Sources/App/Views/Sidebar/SidebarItem.swift:
```diff
@@ -25,6 +25,7 @@ struct SidebarItem: View {
   // MARK: - Body
 
   var body: some View {
+
```
```suggestion
```

davideruisi commented on Sources/App/Views/Common/Buttons/AccentButtonStyle.swift:
```diff
@@ -32,4 +26,11 @@ struct AccentButtonStyle: ButtonStyle {
   }
+
+  func makeBody(configuration: Self.Configuration) -> some View {
+    return configuration.label
```
```suggestion
    configuration.label
```

davideruisi reviewed (approved):
great work 👏 only two minor things
